cli/src/commands/branch.rs

```diff
@@ -301,6 +319,42 @@ fn cmd_branch_create(
     Ok(())
 }
 
+fn cmd_branch_rename(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &BranchRenameArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let repo = workspace_command.repo().as_ref();
+    let old_branch = &args.old;
+    // call clone because .set_local_branch_target takes ownership
+    // of the target_commit, so we need to create a copy that is no
+    // longer owned by the workspace.
+    let target_commit = repo.view().get_local_branch(old_branch).clone();
+    if target_commit.is_absent() {
+        return Err(user_error_with_hint(
+            format!("Branch not found: {old_branch}"),
+            "Target branch name does not exist.",
```

Comment:
nit: just say `"No such branch: {old_branch}"` (without hint)?

I don't think "Target branch name does not exist." adds additional context.

Reply:
Heh! Indeed. Removed.